nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -226,8 +236,21 @@ geometry_msgs::msg::TwistStamped RegulatedPurePursuitController::computeVelocity
   // Transform path to robot base frame
   auto transformed_plan = transformGlobalPlan(pose);
 
-  // Find look ahead distance and point on path and publish
-  const double lookahead_dist = getLookAheadDistance(speed);
+  double lookahead_dist = getLookAheadDistance(speed);  
+
+  // Check for reverse driving
+  if(reverse_driving_)
+  {
+    // Cusp check
+    double orientation_change_dist_= checkOrientation(pose);
```

Comment:
This function name is not good, I have no idea what its doing. Please rename this function to something inline readable to know what's going on.